qdrant_client/local/local_collection.py

```diff
@@ -133,6 +135,33 @@ def close(self) -> None:
         if self.storage is not None:
             self.storage.close()
 
+    def _update_idf_append(self, vector: SparseVector, vector_name: str) -> None:
+        if vector_name not in self.sparse_vectors_idf:
+            self.sparse_vectors_idf[vector_name] = defaultdict(int)
+        for idx in vector.indices:
+            self.sparse_vectors_idf[vector_name][idx] += 1
+
+    def _update_idf_remove(self, vector: SparseVector, vector_name: str) -> None:
+        for idx in vector.indices:
+            self.sparse_vectors_idf[vector_name][idx] -= 1
+
+    @classmethod
+    def _compute_idf(cls, df: int, n: int) -> float:
+        # ((n - df + 0.5) / (df + 0.5) + 1.).ln()
+        return math.log((n - df + 0.5) / (df + 0.5) + 1)
```

Comment:
why do we need to add `+1` ?

I thought that it might be to avoid a case when the expression in the parentheses is equal to 0, however it is not possible since `n > df`